diff --git a/arch/arm/boot/dts/am335x-chiliboard-common.dtsi b/arch/arm/boot/dts/am335x-chiliboard-common.dtsi
index 5c8d86b..956cc4b 100644
--- a/arch/arm/boot/dts/am335x-chiliboard-common.dtsi
+++ b/arch/arm/boot/dts/am335x-chiliboard-common.dtsi
@@ -36,6 +36,13 @@
 &am33xx_pinmux {
 	pinctrl-names = "default";
 
+	uart0_pins: pinmux_uart0_pins {
+		pinctrl-single,pins = <
+			0x170 (PIN_INPUT_PULLUP | MUX_MODE0)	/* uart0_rxd.uart0_rxd */
+			0x174 (PIN_OUTPUT_PULLDOWN | MUX_MODE0) /* uart0_txd.uart0_txd */
+		>;
+	};
+
 	cpsw_default: cpsw_default {
 		pinctrl-single,pins = <
 			0x10c (PIN_INPUT_PULLDOWN | MUX_MODE1)	/* mii1_crs.rmii1_crs_dv */
@@ -106,6 +113,12 @@
 
 };
 
+&uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_pins>;
+	status = "okay";
+};
+
 &mac {
 	slaves = <1>;
 	pinctrl-names = "default", "sleep";
diff --git a/arch/arm/boot/dts/am335x-chilisom-gcs2.dtsi b/arch/arm/boot/dts/am335x-chilisom-gcs2.dtsi
index 0992e5d..e8a3456 100644
--- a/arch/arm/boot/dts/am335x-chilisom-gcs2.dtsi
+++ b/arch/arm/boot/dts/am335x-chilisom-gcs2.dtsi
@@ -33,13 +33,6 @@
 		>;
 	};
 
-	uart0_pins: pinmux_uart0_pins {
-		pinctrl-single,pins = <
-			0x170 (PIN_INPUT_PULLUP | MUX_MODE0)	/* uart0_rxd.uart0_rxd */
-			0x174 (PIN_OUTPUT_PULLDOWN | MUX_MODE0)	/* uart0_txd.uart0_txd */
-		>;
-	};
-
 	nandflash_pins_default: nandflash_pins_default {
 		pinctrl-single,pins = <
 			0x0 (PIN_INPUT_PULLUP | MUX_MODE0)	/* gpmc_ad0.gpmc_ad0 */
@@ -79,12 +72,6 @@
 	};
 };
 
-&uart0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_pins>;
-	status = "okay";
-};
-
 &i2c0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c0_pins>;
--
libgit2 0.21.0
